### PR TITLE
feat(upload files): upload avatar is ready for test

### PR DESCRIPTION
**Pull request by DiLiUA**
- developed functionality for upload avatar;
  - refacrored functionality for upload poster;

### CODE REVIEW

File: routes/upload-files.ts [hunk 1]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
-function uploadPoster(req: FileUploaderRequest, res: any) {
+async function uploadAvatar(req: Request, res: Response): Promise<void> {
+  const file = req.body.fileBody;
+  const headers = req.headers;
+  const filename = headers.filename || new Date().getTime().toString();
+  const pathName = `uploads/avatars/${filename}`;
+  const allowedMimeTypes = ['image/gif', 'image/jpeg', 'image/pjpeg', 'image/x-png', 'image/png', 'image/svg+xml'];
+
+  if (allowedMimeTypes.indexOf(headers['content-type']) === -1) {
+    const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:51]: use HTTPStatus

File: routes/upload-files.ts [hunk 2]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
+
+    res.status(status).send({message: 'Unsupported type of file'});
+  }
+
+  const savedFile = await saveFile(`./${pathName}`, file);
+
+  if (!savedFile.isSuccess) {
+    const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:59]: use HTTPStatus

File: routes/upload-files.ts [hunk 3]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
+}
+
+async function uploadPoster(req: FileUploaderRequest, res: Response): Promise<void> {
   const files = req.files;
   const body = req.body;
   const allowedMimeTypes = ['image/gif', 'image/jpeg', 'image/pjpeg', 'image/x-png', 'image/png', 'image/svg+xml'];
 
   if (!files.length) {
-    return res.json({success: false, data: null, error: 'Field is required'});
+    const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:87]: use HTTPStatus

File: routes/upload-files.ts [hunk 4]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
+
+    res.status(status).send({message: 'Field is required'});
   }
 
   files.forEach(file => {
     if (allowedMimeTypes.indexOf(file.mimetype) === -1) {
-      return res.json({success: false, data: null, error: 'Unsupported type of file'});
+      const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:94]: use HTTPStatus

File: routes/upload-files.ts [hunk 5]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
+    const pathName = `uploads/posters/${uniqName}`;
 
-    fs.writeFile(`./uploads/posters/${uniqName}`, file.buffer, (err: Error) => {
-      if (err) {
-        return res.json({success: !err, data: null, err});
-      }
+    saveFile(`./${pathName}`, file.buffer)
+      .then(savedFile => {
+        if (!savedFile.isSuccess) {
+          const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:105]: use HTTPStatus

File: routes/upload-files.ts [hunk 6]
```diff
@@ -25,41 +40,124 @@ interface FileUploaderRequest extends Request {
...
 
-          return res.json({success: error, data: 'success'});
-        });
+        updateEvent(updateParams)
+          .then(() => {
+            return res.json({success: true, data: 'success'});
+          });
+      })
+      .catch(err => {
+        const status = 500;
```
Comment (Ugmaxie) [inline routes/upload-files.ts:123]: use HTTPStatus

**Review by Ugmaxie — CHANGES_REQUESTED** [review]

File: routes/upload-files.ts (hunk 1 above)
Comment (DiLiUA) [reply routes/upload-files.ts:51]: I'll do it for all project  in a separate PR

File: routes/upload-files.ts (hunk 2 above)
Comment (DiLiUA) [reply routes/upload-files.ts:59]: I'll do it for all project  in a separate PR

File: routes/upload-files.ts (hunk 3 above)
Comment (DiLiUA) [reply routes/upload-files.ts:87]: I'll do it for all project  in a separate PR

File: routes/upload-files.ts (hunk 4 above)
Comment (DiLiUA) [reply routes/upload-files.ts:94]: I'll do it for all project  in a separate PR

File: routes/upload-files.ts (hunk 5 above)
Comment (DiLiUA) [reply routes/upload-files.ts:105]: I'll do it for all project  in a separate PR

File: routes/upload-files.ts (hunk 6 above)
Comment (DiLiUA) [reply routes/upload-files.ts:123]: I'll do it for all project  in a separate PR

**Review by Ugmaxie — APPROVED** [review]

**Pull request merged**